src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/CastExpressionSyntaxExtensions.vb

```diff
@@ -10,21 +10,127 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
+            End If
+
+            If trailingTrivia.Count > 0 Then
+                Dim newTrailingTrivia As New List(Of SyntaxTrivia)
+                Dim foundEOL As Boolean = False
+                For i As Integer = 0 To trailingTrivia.Count - 1
+                    Dim trivia As SyntaxTrivia = trailingTrivia(i)
+                    Dim nextTrivia As SyntaxTrivia = GetNextTrivia(trailingTrivia, i, 1)
+                    Select Case trivia.Kind
+                        Case SyntaxKind.WhitespaceTrivia
+                            Select Case nextTrivia.Kind
+                                Case SyntaxKind.WhitespaceTrivia
+                                    trailingTrivia(i + 1) = AdjustWhitespace(trivia, nextTrivia, GetNextTrivia(trailingTrivia, i, 2).IsKind(SyntaxKind.LineContinuationTrivia))
+                                Case SyntaxKind.EndOfLineTrivia
+                                    ' skip Whitespace before EOL
+                                Case SyntaxKind.LineContinuationTrivia
+                                    If GetNextTrivia(trailingTrivia, i, 2).IsKind(SyntaxKind.WhitespaceTrivia) Then
+                                        trailingTrivia(i + 2) = trivia
+                                        trivia = SyntaxFactory.Space
+                                    End If
+                                    newTrailingTrivia.Add(trivia)
+                                Case SyntaxKind.CommentTrivia
+                                    newTrailingTrivia.Add(trivia)
+                                Case Else
+                                    newTrailingTrivia.Add(trivia)
+                            End Select
+                        Case SyntaxKind.EndOfLineTrivia
+                            If Not foundEOL Then
+                                newTrailingTrivia.Add(trivia)
+                                foundEOL = True
+                            End If
+                        Case SyntaxKind.LineContinuationTrivia
+                            newTrailingTrivia.Add(trivia)
+                        Case SyntaxKind.CommentTrivia
+                            newTrailingTrivia.Add(trivia)
+                            foundEOL = False
+                        Case Else
+                            foundEOL = False
+                            newTrailingTrivia.Add(trivia)
+                    End Select
+                Next
+                trailingTrivia = newTrailingTrivia
+            End If
+            Dim resultNode = innerNode.With(leadingTrivia, trailingTrivia)
 
             resultNode = SimplificationHelpers.CopyAnnotations(castNode, resultNode)
 
             Return resultNode
 
         End Function
+
+        Private Function GetNextTrivia(trailingTrivia As List(Of SyntaxTrivia), index As Integer, lookaheadCount As Integer) As SyntaxTrivia
+            Return If(index < trailingTrivia.Count - lookaheadCount, trailingTrivia(index + lookaheadCount), Nothing)
```

Comment:
Method name does not indicate at all that it has out of bounds handling semantics.

Reply:
FIxed